drivers/hmis/app/models/hmis/strict_integer.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class Hmis::StrictInteger < ActiveModel::Type::Integer
+  INT_RGX = /\A-?\d+(\.0+)?\z/
```

Comment:
This is a different regex than you had proposed @ttoomey:
- this one `/\A-?\d+(\.0+)?\z/` allows '123.0' but rejects '123.45'
- your proposal `/\A-?\d+(\.\d*[1-9])?\z|\A-?\d+\.0+\z/` accepts both and casts them to 123